server/models/User.js

```diff
@@ -292,10 +298,45 @@ User.getCounts = async function (teamId: string) {
   return {
     active: parseInt(counts.activeCount),
     admins: parseInt(counts.adminCount),
+    viewers: parseInt(counts.viewerCount),
     all: parseInt(counts.count),
     invited: parseInt(counts.invitedCount),
     suspended: parseInt(counts.suspendedCount),
   };
 };
 
+User.prototype.demote = async function (teamId: string, to: string) {
+  const res = await User.findAndCountAll({
+    where: {
+      teamId,
+      isAdmin: true,
+      id: {
+        [Op.ne]: this.id,
+      },
+    },
+    limit: 1,
+  });
+
+  if (res.count >= 1) {
+    if (to === "member") {
+      return this.update({ isAdmin: false, isViewer: false });
+    } else if (to === "viewer") {
+      return this.update({ isAdmin: false, isViewer: true });
+    }
+  } else {
+    throw new ValidationError("At least one admin is required");
+  }
+};
+
+User.prototype.makeAdmin = async function () {
```

Comment:
Should we convert this to `promote` ? The asymmetry is awkward as I'm not sure how we would make a `viewer` -> `member` with this API?

Reply:
The `users.promote` endpoint is only to used to make the user an `admin`. `users.demote` is used to make a user `viewer` -> `member` or `member` to `viewer`. 

I know this is awkward as well. What do you think about having a single endpoint? That will change the rank of the current user based on the `to` parameter in the `req` body. 

`to = ['admin','member','viewer']`

Reply:
It's tempting… but it's probably going to make life easier to keep the endpoints aligned with the policies, otherwise you need to calculate if the `to` parameter is promoting or demoting. Plus it enables a roll out without a breaking API change.

Reply:
I was conscious of the breaking API change and that's why I didn't put forward the idea before.

Reply:
In conclusion, I'm renaming the function to `promote`.